### ISSUE TITLE
Do we really want to prevent opening multiple times the same class?

### ISSUE THREAD

**Issue by echebbi** [gemoc/ale-lang#161]
## Context

Opening several times the same class results in an error:

![image](https://user-images.githubusercontent.com/25926433/85927511-cfe20700-b8a6-11ea-9103-be51e431e4a8.png)

However I don't understand the rational behind it. To solve the issue it seems that we have to use the `extends` keyword as follows:

```kotlin
open class StateMachine {

}

open class StateMachine extends StateMachine {

}
```
which doesn't make any sense to me.

I think we should allow it, but maybe add a marker on the ruler to warn about the class already being open & to provide quick access to the other location.

## Notes

For the record, this validation is done [here](https://github.com/gemoc/ale-lang/blob/265f9467054f2266df76215dd9b195d57c6ca751/plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/validation/OpenClassValidator.java#L103-L110).

**Comment by dvojtise** [comment]
I agree we may want to open it several times but most probably in 2 different ale files in order to create 2 aspects. Each contributing some new features for different purposes. The fact that we cannot name the aspects would be counter balanced by the name of tmeach ale file. Even if not best practice (need for doc on the each aspect to clarify the usage of each of them) in some case we may wish to put everything in a single file

**Comment by echebbi** [comment]
> I agree we may want to open it several times but most probably in 2 different ale files in order to create 2 aspects. Each contributing some new features for different purposes

I think that's important to support this scenario. We may also decide that an EClass shouldn't be open more than once in a single source to enforce some "best" practices but in that case we should also forbid this strange `extends`-workaround.

Remaining tasks:
 - Allow to open the same EClass when the aspects are defined in different source files
 - Decide whether we want to allow or forbid to open an EClass multiple times in the same source file
